### PR TITLE
spm: Mark SPM as deprecated

### PR DESCRIPTION
**Pull request by joerchan**
Mark SPM configuration as deprecated and add __deprecate attribute
to the SPM secure API functions.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

### CODE REVIEW

**Review by SebastianBoe — APPROVED** [review]

**Review by oyvindronningstad — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -135,6 +135,11 @@ RF Front-End Modules
   The :ref:`MPSL library <nrfxlib:mpsl>` is linked into the build without initialization.
   You cannot use other MPSL features when this option is enabled.
 
+Secure Partition Manager (SPM)
+==============================
+
+* The Secure Partition Manager (SPM), kconfig option :kconfig:option:`CONFIG_SPM`, is now deprecated. It is replaced by Trusted Firmware-M (TF-M) as the supported solution.
```
Comment (oyvindronningstad) [inline doc/nrf/releases/release-notes-changelog.rst:141]: This could have a link to TF-M (user guide or kconfig or something)

**Comment by SebastianBoe** [comment]
rebased to pass CI

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (SebastianBoe) [reply doc/nrf/releases/release-notes-changelog.rst]: done. Cargo cult copied from elsewhere in this file as I didn't know how to make the link.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -135,6 +135,11 @@ RF Front-End Modules
   The :ref:`MPSL library <nrfxlib:mpsl>` is linked into the build without initialization.
   You cannot use other MPSL features when this option is enabled.
 
+Secure Partition Manager (SPM)
+==============================
+
+* The Secure Partition Manager (SPM), kconfig option :kconfig:option:`CONFIG_SPM`, is now deprecated. It is replaced by Trusted Firmware-M (TF-M) as the supported solution. For information about TF-M see :ref:`TF-M <ug_tfm>`.
```
Comment (mia-ko) [inline doc/nrf/releases/release-notes-changelog.rst:141]:
```suggestion
* The Secure Partition Manager (SPM), the Kconfig option :kconfig:option:`CONFIG_SPM`, is now deprecated.
  It is replaced by Trusted Firmware-M (TF-M) as the supported solution.
  For information about TF-M see :ref:`TF-M <ug_tfm>`.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 2 above)
Comment (Vge0rge) [reply doc/nrf/releases/release-notes-changelog.rst]: @mia-ko  please approve when you verify the changes.

**Review by oyvindronningstad — APPROVED** [review]

**Comment by Vge0rge** [comment]
> Should the readme file of the SPM sample also state that the SPM is deprecated?

@mia-ko  I just updated the readme file to include the deprecation in the warning section. Please review

File: samples/spm/README.rst [hunk 3]
```diff
@@ -11,7 +11,8 @@ The Secure Partition Manager sample provides a reference use of the System Prote
 This firmware sets up an nRF device with Trusted Execution (|trusted_execution|) so that it can run user applications in the non-secure domain.
 
 .. note::
-   SPM is an alternative for using the Trusted Firmware-M (TF-M). See :ref:`ug_tfm`.
+   SPM is now DEPRECATED and will be removed in a future version of the SDK.
```
Comment (mia-ko) [inline samples/spm/README.rst:14]:
```suggestion
   SPM is deprecated as of |NCS| v2.1.0 and will be removed in a future version of the SDK.
```
We should probably mention the version of NCS it is deprecated in, right?

Comment (Vge0rge) [reply samples/spm/README.rst]: Done

File: samples/spm/README.rst [hunk 4]
```diff
@@ -11,7 +11,8 @@ The Secure Partition Manager sample provides a reference use of the System Prote
 This firmware sets up an nRF device with Trusted Execution (|trusted_execution|) so that it can run user applications in the non-secure domain.
 
 .. note::
-   SPM is an alternative for using the Trusted Firmware-M (TF-M). See :ref:`ug_tfm`.
+   SPM is deprecated as of |NCS| v2.1.0 and will be removed in a future version of the SDK.
+   Trusted Firmware-M (TF-M) :ref:`ug_tfm` will replace SPM as the trusted execution solution.
```
Comment (mia-ko) [inline samples/spm/README.rst:15]:
```suggestion
   :ref:`Trusted Firmware-M (TF-M) <ug_tfm>` will replace SPM as the trusted execution solution.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -138,6 +138,13 @@ RF Front-End Modules
   The :ref:`MPSL library <nrfxlib:mpsl>` is linked into the build without initialization.
   You cannot use other MPSL features when this option is enabled.
 
+Secure Partition Manager (SPM)
+==============================
+
+* The Secure Partition Manager (SPM), the Kconfig option :kconfig:option:`CONFIG_SPM`, is now deprecated.
+  It is replaced by the Trusted Firmware-M (TF-M) as the supported solution.
+  For information about the TF-M, see :ref:`TF-M <ug_tfm>`.
```
Comment (mia-ko) [inline doc/nrf/releases/release-notes-changelog.rst:146]:
```suggestion
* The Secure Partition Manager (SPM), the Kconfig option :kconfig:option:`CONFIG_SPM`, is deprecated.
  It is replaced by the Trusted Firmware-M (TF-M) as the supported trusted execution solution.
  See :ref:`ug_tfm` for more information about the TF-M.
```

**Review by mia-ko — APPROVED** [review]

**Review by mia-ko — COMMENTED** [review]
Just wondering if the readme file for the secure services sample also should say something about the SPM being deprecated... Sorry for not pointing that out before. But that would be the last thing from me, I promise :)

**Comment by joerchan** [comment]
> Just wondering if the readme file for the secure services sample also should say something about the SPM being deprecated... Sorry for not pointing that out before. But that would be the last thing from me, I promise :)

Will delete that sample instead. No need to have a sample for a deprecated feature. Will keep the tests though.

**Comment by rlubos** [comment]
Ignoring unstable `fw-update` plan.

**Pull request merged**